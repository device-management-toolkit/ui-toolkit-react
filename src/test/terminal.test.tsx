/*********************************************************************
* Copyright (c) Intel Corporation 2019
* SPDX-License-Identifier: Apache-2.0
**********************************************************************/
import Term, { IPropTerminal } from '../reactjs/SerialOverLAN/Terminal'
import { Terminal } from 'xterm'

import React from 'react'
import { shallow } from 'enzyme'

describe('Test Term class', () => {
  it('Test render() in Term class', () => {
    const t = new Terminal({
      cursorStyle: 'block',
      fontWeight: 'bold',
      rows: 30,
      cols: 100
    })

    const handleKeyPress = (domEvent): any => jest.fn()
    const handleKeyDownPress = (domEvent): any => jest.fn()
    // Initialization of ConnectProps
    const terminal: IPropTerminal = {
      handleKeyPress: (handleKeyPress),
      xterm: (t),
      handleKeyDownPress: (handleKeyDownPress)
    }

    const term = shallow(<Term { ...terminal } />)

    // Output
    console.log(term.debug())
    expect(term).toMatchSnapshot()
  })
})
