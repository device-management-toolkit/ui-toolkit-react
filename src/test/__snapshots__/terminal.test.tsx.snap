// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Term class Test render() in Term class 1`] = `ShallowWrapper {}`;
